contracts/validator-manager/PoSValidatorManager.sol

```diff
@@ -491,22 +495,28 @@ abstract contract PoSValidatorManager is
         if (delegator.status != DelegatorStatus.Active) {
             revert InvalidDelegatorStatus(delegator.status);
         }
+
         // Only the delegation owner or parent validator can end the delegation.
         if (delegator.owner != _msgSender()) {
             // Validators can only remove delegations after the minimum stake duration has passed.
-            if ($._posValidatorInfo[validationID].owner == _msgSender()) {
-                if (
-                    block.timestamp
-                        < validator.startedAt + $._posValidatorInfo[validationID].minStakeDuration
-                ) {
-                    revert MinStakeDurationNotPassed(uint64(block.timestamp));
-                }
-            } else {
+            if ($._posValidatorInfo[validationID].owner != _msgSender()) {
                 revert UnauthorizedOwner(_msgSender());
             }
+
+            if (
+                block.timestamp
+                    < validator.startedAt + $._posValidatorInfo[validationID].minStakeDuration
+            ) {
+                revert MinStakeDurationNotPassed(uint64(block.timestamp));
```

Comment:
I think this check would make more sense in `initializeDelegatorRegistration`, that way the validator's weight is guaranteed not to change until the churn period has elapsed.

Reply:
This check is for Validator's that want to end a delegation attached to them. They can only end the delegation after their `minStakeDuration` has elapsed. I only refactored the nesting for these checks, I didn't change any functionality here.

Reply:
Whoops, thanks for clarifying.